doc/user_guide/jupyter_chart.rst

```diff
@@ -0,0 +1,403 @@
+.. _user-guide-jupyterchart:
+
+JupyterChart
+============
+The ``JupyterChart`` class, introduced in Vega-Altair 5.1, makes it possible to update charts
+after they have been displayed and access the state of :ref:`user-guide-interactions` from Python.
+
+Supported Environments
+----------------------
+``JupyterChart`` is a `Jupyter Widget <https://ipywidgets.readthedocs.io/en/latest/>`_ built
+on the `AnyWidget <https://anywidget.dev/>`_ library. As such, it's compatible with development
+environments and dashboard toolkits that support third party Jupyter Widgets.
+Tested environments include:
+
+* Classic Jupyter Notebook
+* JupyterLab
+* Visual Studio Code
+* Google Colab
+* Voila
+
+.. note::
+    If you try ``JupyterChart`` in another environment that supports Jupyter Widgets,
+    `let us know how it goes <https://github.com/altair-viz/altair/issues>`_ so that we can keep
+    this list up to date.
+
+Basic Usage
+-----------
+To create a ``JupyterChart``, pass a regular ``Chart`` instance to the ``alt.JupyterChart``
+constructor. The chart will be displayed automatically if the last expression in a notebook
+cell evaluates to a ``JupyterChart`` instance. For example:
+
+.. code-block:: python
+
+    import altair as alt
+    import pandas as pd
+
+    source = pd.DataFrame({
+        'a': ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I'],
+        'b': [28, 55, 43, 91, 81, 53, 19, 87, 52]
+    })
+
+    chart = alt.Chart(source).mark_bar().encode(
+        x='a',
+        y='b'
+    )
+
+    jchart = alt.JupyterChart(chart)
+    jchart
+
+.. image:: https://github.com/altair-viz/altair/assets/15064365/59cb2484-ad51-461a-8353-e1d4fa5f6bb6
+  :alt: Bar chart with letters A through I on the x-axis
+
+
+Updating Charts
+---------------
+The ``JupyterChart``'s ``chart`` property can be assigned to a new chart instance, and the new chart
+will immediately be displayed in place of the old one.
+
+.. code-block:: python
+
+    jchart.chart = chart.mark_bar(color="crimson", cornerRadius=10)
+
+.. raw:: html
+
+    <video controls>
+      <source src="https://github.com/altair-viz/altair/assets/15064365/21d580ef-4dec-4687-a765-9b6d6d84c61f">
```

Comment:
I think we should place this in the `static` folder, so the docs are self-contained in a single folder, so we can facilitate scrapers for offline documentation software. I'm not sure on this one btw.

Reply:
Done in e097995767d635137f232735b67e4addde42294c.

Note that this does increase the repository size by ~14MB. Not a huge deal from my POV, but something to keep an eye one.

Reply:
We could apply video compression to serve users who are residing in low-bandwidth locations.

Reply:
I wouldn't want to degrade the resolution for all users.  One option would be to upload these to the VegaFusion YouTube channel (or we could make an altair-viz YouTube channel) and embed the videos from YouTube. YouTube handles serving variable resolution based on bandwidth, and this would keep the video files out of the Altair repo.

I don't know how offline documentation systems work, and whether they are already set up to handle embedded youtube videos (which I imagine being pretty common). Do we know of any offline documentation systems that currently support scraping the Vega-Altair docs?

Reply:
I know that @capac has been doing this for different Altair releases for [Dash for MacOS](https://kapeli.com/dash) in the past, https://github.com/altair-viz/altair/issues/3028#issuecomment-1517885924.